esx_service/utils/auth_api.py

```diff
@@ -631,10 +637,20 @@ def _tenant_access_set(name, datastore, allow_create=None, volume_maxsize_in_MB=
         error_info = ErrorInfo(err_code, err_msg)
         return error_info
 
+    allow_create_val = str(allow_create)
+    allow_create_val = allow_create_val.lower()
```

Comment:
Using ```allow_create_val = str(allow_create).lower()```
Preserving original allow_create value to log error and return it in case of invalid input.